test/integration/relation.js

```diff
@@ -260,6 +280,117 @@ module.exports = function(Bookshelf) {
         equal(_knex.toString(), sql);
       });
 
+      it('should handle a hasOne relation with explicit foreignKeyTarget', function() {
+        const base = new Locale({ id: 1, isoCode: 'en' });
+        const relation = base.translation();
+        const _knex = relation.query();
+        const relatedData = relation.relatedData;
+
+        // Base
+        equal(relatedData.type, 'hasOne');
+        equal(relatedData.target, Translation);
+        equal(relatedData.targetTableName, 'translations');
+        equal(relatedData.targetIdAttribute, 'id');
+        equal(relatedData.foreignKey, 'locale');
+        equal(relatedData.foreignKeyTarget, 'isoCode');
+
+        // Init
+        equal(relatedData.parentId, 1);
+        equal(relatedData.parentTableName, 'locales');
+        equal(relatedData.parentIdAttribute, 'id');
+        equal(relatedData.parentFk, 'en');
+
+        // init the select constraints
+        relatedData.selectConstraints(_knex, {});
+
+        equal(_knex.toString(), 'select `translations`.* from `translations` where `translations`.`locale` = \'en\' limit 1');
```

Comment:
It goes for all relation types: `hasOne`, `belongsTo`, ´hasMany´ & `belongsToMany`.